components/procore/sources/common.js

```diff
@@ -0,0 +1,70 @@
+const procore = require("../procore.app.js");
+
+module.exports = {
+  dedupe: "unique",
+  props: {
+    procore,
+    db: "$.service.db",
+    http: "$.interface.http",
+    company: { propDefinition: [procore, "company"] },
+    project: {
+      propDefinition: [procore, "project", (c) => ({ company: c.company })],
+    },
+  },
+  methods: {
+    getEventTypes() {
+      return ["create", "update", "delete"];
+    },
+  },
+  hooks: {
+    async activate() {
+      const hook = await this.procore.createHook(
+        this.http.endpoint,
+        this.company,
+        this.project
+      );
+      this.db.set("hookId", hook.id);
+      // create hook triggers
+      eventTypes = this.getEventTypes();
+      resourceName = this.getResourceName();
```

Comment:
Minor: we might want to add a default implementation of `getResourceName`, even if it's dummy or if child classes must implement it:
```js
module.exports = {
  methods: {
    getResourceName() {
      throw new Error("getResourceName is not implemented");
    },
  },
};
```

This mainly helps with readability, so it's a minor request at this point.